src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/RoundTripIslSpec.groovy

```diff
@@ -288,4 +289,156 @@ round trip latency rule is removed on the dst switch"() {
+        } ?: assumeTrue("Wasn't able to find a suitable link", false)
+
+        antiflap.portDown(roundTripIsl.srcSwitch.dpId, roundTripIsl.srcPort)
+        Wrappers.wait(WAIT_OFFSET) {
+            def links = northbound.getAllLinks()
+            assert islUtils.getIslInfo(links, roundTripIsl).get().state == FAILED
+            assert islUtils.getIslInfo(links, roundTripIsl.reversed).get().state == FAILED
+        }
+        def portIsUp = false
+        northbound.deleteLink(islUtils.toLinkParameters(roundTripIsl))
+        Wrappers.wait(WAIT_OFFSET) {
+            def links = northbound.getAllLinks()
+            assert !islUtils.getIslInfo(links, roundTripIsl).present
+        }
+
+        when: "Disable portDiscovery on the srcPort"
+        northboundV2.updatePortProperties(roundTripIsl.srcSwitch.dpId, roundTripIsl.srcPort,
+                new PortPropertiesDto(discoveryEnabled: false))
+        def portDiscoveryIsEnabledOnSrcPort = false
+
+        and: "Revive the ISL back (bring switch port up)"
+        antiflap.portUp(roundTripIsl.srcSwitch.dpId, roundTripIsl.srcPort)
+        portIsUp = true
+
+        then: "The ISL is rediscovered"
+        Wrappers.wait(discoveryInterval + WAIT_OFFSET) {
+            def links = northbound.getAllLinks()
+            assert islUtils.getIslInfo(links, roundTripIsl).get().state == DISCOVERED
+            assert islUtils.getIslInfo(links, roundTripIsl.reversed).get().state == DISCOVERED
+        }
+
+        and: "The src/dst switches are valid"
+        //https://github.com/telstra/open-kilda/issues/3906
+//        [roundTripIsl.srcSwitch, roundTripIsl.dstSwitch].each {
+//            def validateInfo = northbound.validateSwitch(it.dpId).rules
+//            assert validateInfo.missing.empty
+//            assert validateInfo.excess.empty
+//            assert validateInfo.misconfigured.empty
+//        }
+
+        when: "Disable portDiscovery on the dstPort"
+        northboundV2.updatePortProperties(roundTripIsl.dstSwitch.dpId, roundTripIsl.dstPort,
+                new PortPropertiesDto(discoveryEnabled: false))
+        def portDiscoveryIsEnabledOnDstPort = false
+
+        then: "The ISL is failed"
+        Wrappers.wait(WAIT_OFFSET) {
+            def links = northbound.getAllLinks()
+            assert islUtils.getIslInfo(links, roundTripIsl).get().state == FAILED
+            assert islUtils.getIslInfo(links, roundTripIsl.reversed).get().state == FAILED
+        }
+
+        when: "Delete the ISL without the 'force' option"
+        northbound.deleteLink(islUtils.toLinkParameters(roundTripIsl))
+
+        then: "The ISL is deleted"
+        Wrappers.wait(WAIT_OFFSET) {
+            def links = northbound.getAllLinks()
+            assert !islUtils.getIslInfo(links, roundTripIsl).present
+        }
+
+        cleanup:
+        !portDiscoveryIsEnabledOnSrcPort && northboundV2.updatePortProperties(roundTripIsl.srcSwitch.dpId,
```

Comment:
If the assumption fails on L371, this code will still try to run. Need a better solution here and below

Reply:
fixed